jib-core/src/main/java/com/google/cloud/tools/jib/builder/BuildImageSteps.java

```diff
@@ -159,6 +160,19 @@ public void run()
                               entrypoint),
                           listeningExecutorService);
 
+          timer2.lap("Setting up container configuration push");
+          // Pushes the container configuration.
+          ListenableFuture<ListenableFuture<BlobDescriptor>>
+              pushContainerConfigurationFutureFuture =
+                  Futures.whenAllSucceed(buildContainerConfigurationFutureFuture)
+                      .call(
+                          new PushContainerConfigurationStep(
+                              buildConfiguration,
+                              authenticatePushFuture,
+                              buildContainerConfigurationFutureFuture.get(),
```

Comment:
This call to get would be executed before anything is run, which would cause race conditions and blocking. Any `get` for a `Future` should be called `call` method of the step that uses it and should be wrapped in a `NonBlockingFutures.get` for busy-wait safety (it makes sure that the `get` does not block).